pkg/workloads/console/controller.go

```diff
@@ -166,15 +168,21 @@ func (r *reconciler) Reconcile() (res k8rec.Result, err error) {
 	}
 
 	// Requeue reconciliation if the status may change
-	switch r.console.Status.Phase {
-	case workloadsv1alpha1.ConsolePending:
+	switch {
+	case r.console.Pending():
 		res = requeueAfterInterval(r.logger, time.Second)
-	case workloadsv1alpha1.ConsoleRunning:
+	case r.console.Running():
 		res = requeueForExpiration(r.logger, r.console.Status)
+	case r.console.Stopped():
+		// Requeue for when the console needs to be deleted
+		// In the future we could allow this to be configured in the template
+		res = requeueAfterInterval(r.logger, ConsoleTTL)
 	}
 
-	// TODO:
-	//   GC the terminated console if necessary
+	if r.console.Stopped() && time.Now().Sub(r.console.Status.ExpiryTime.Time) > ConsoleTTL {
+		r.logger.Log("event", EventDeleted, "resource", Console)
+		r.client.Delete(r.ctx, r.console)
```

Comment:
I think you need to set a propagation policy on this, so that it cleans up the dependent resources?

Reply:
Good point!